database/database.go

```diff
@@ -45,7 +52,7 @@ var (
 	// mongoReadPreference defines the DB's read preference. The options are:
 	// primary, primaryPreferred, secondary, secondaryPreferred, nearest.
 	// See https://docs.mongodb.com/manual/core/read-preference/
-	mongoReadPreference = "nearest"
+	mongoReadPreference = "primary"
```

Comment:
This change is required for transactions. It's unfortunate but if we want transactions, we need to read from the primary.